model/__init__.py

```diff
@@ -0,0 +1,533 @@
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore", category=SAWarning)
+            engine, connection = cls.initialize(url)
+        session = Session(connection)
+        if initialize_data:
+            session = cls.initialize_data(session)
+        return session
+
+    @classmethod
+    def initialize_data(cls, session, set_site_configuration=True):
+        # Create initial content.
+        from datasource import DataSource
+        from classification import Genre
+        from licensing import DeliveryMechanism
+        list(DataSource.well_known_sources(session))
+
+        # Load all existing Genre objects.
+        Genre.populate_cache(session)
+
+        # Create any genres not in the database.
+        for g in classifier.genres.values():
+            # TODO: On the very first startup this is rather expensive
+            # because the cache is invalidated every time a Genre is
+            # created, then populated the next time a Genre is looked
+            # up. This wouldn't be a big problem, but this also happens
+            # on setup for the unit tests.
+            Genre.lookup(session, g, autocreate=True)
+
+        # Make sure that the mechanisms fulfillable by the default
+        # client are marked as such.
+        for content_type, drm_scheme in DeliveryMechanism.default_client_can_fulfill_lookup:
+            mechanism, is_new = DeliveryMechanism.lookup(
+                session, content_type, drm_scheme
+            )
+            mechanism.default_client_can_fulfill = True
+
+        # If there is currently no 'site configuration change'
+        # Timestamp in the database, create one.
+        timestamp, is_new = get_one_or_create(
+            session, Timestamp, collection=None,
+            service=Configuration.SITE_CONFIGURATION_CHANGED,
+            create_method_kwargs=dict(timestamp=datetime.datetime.utcnow())
+        )
+        if is_new:
+            from listeners import site_configuration_has_changed
+            site_configuration_has_changed(session)
+        session.commit()
+
+        # Return a potentially-new Session object in case
+        # it was updated by cls.update_timestamps_table
+        return session
+
+def production_session():
+    url = Configuration.database_url()
+    if url.startswith('"'):
+        url = url[1:]
+    logging.debug("Database url: %s", url)
+    _db = SessionManager.session(url)
+
+    # The first thing to do after getting a database connection is to
+    # set up the logging configuration.
+    #
+    # If called during a unit test, this will configure logging
+    # incorrectly, but 1) this method isn't normally called during
+    # unit tests, and 2) package_setup() will call initialize() again
+    # with the right arguments.
+    from ..log import LogConfiguration
+    LogConfiguration.initialize(_db)
+    return _db
+
+from admin import (
+    Admin,
+    AdminRole,
+)
+from background import (
```

Comment:
I don't get the name of this module. Did we not use 'coverage' because we already have a coverage.py?

Reply:
That's right, but we can change it now that we're not messing with `sys.path` anymore. There's no chance of confusing `coverage` with `model.coverage`